src/HCubature.jl

```diff
@@ -47,20 +47,36 @@ cubrule(::Val{0}, ::Type{T}) where {T} = Trivial()
 countevals(::Trivial) = 1
 
 """
-    alloc_buf(;dimension[, domain_type, range_type, error_type])
+    hcubature_buffer(;dimension[, domain_type, range_type, error_type])
 
 Allocate a buffer that can be used in calls to [`hcubature`](@ref).
```

Comment:
Docstring needs updating?

Reply:
I thought I updated the doctoring in 29e9222. Still something missing? Also, should I squash the commits?

Reply:
No need to squash the commits.   (Github will do that for you.)

Reply:
It would be good to explicitly document that:

1. The arguments `f, a, b; norm` are the same as those passed to `hcubature`
2. The resulting buffer can be re-used for any integral as long as `a, b` are of the same type and the return value of `f` is of the same type, even if it is different *values* of the endpoints and a different function.
3. `hcubature_buffer` will evaluate `f` at a few points.  This is fine because the function is only really useful if (a) `f` is quite cheap (so that buffer allocation is significant) and (b) you're going to re-use the buffer for many integrals.

Reply:
Done in 180b87d. Since the fact that `hcubature_buffer` evaluates `f` a few times is an implementation detail, I decided not to document that. I added instead a rough discussion on when I believe pre-allocation may be useful.